### PR TITLE
samples: nrf9160: memfault: log updates

### PR DESCRIPTION
**Pull request by jhn-nordic**
Updated logging configuration and documentation to reflect
changes in Memfault SDK

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

### CODE REVIEW

**Review by simensrostad — APPROVED** [review]

**Comment by simensrostad** [comment]
@divipillai Needs a doc review, thanks!

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -331,6 +331,12 @@ nRF9160 samples
     * CoAP max message size is set to 1280 by default.
     * Number of SenML CBOR records is set to a higher value to cope with data exchange after registration with Coiote server.
 
+* :ref:`memfault_sample` sample:
+
+  * Updated:
+
+    * Updated sample to reflect changes with logging in Memfault SDK
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:338]:
```suggestion
* :ref:`memfault_sample` sample:

  * Updated the sample to reflect changes in logging to the `Memfault SDK`_.
```

Comment (jhn-nordic) [reply doc/nrf/releases/release-notes-changelog.rst]: Fixed

**Review by balaji-nordic — APPROVED** [review]
Tested and found it working as expected. 👍

**Review by divipillai — APPROVED** [review]

**Pull request merged**